File: frontend/src/types/task.ts
export type TaskPriority = 'low' | 'medium' | 'high';
export type TaskStatus = 'todo' | 'doing' | 'done';

export type Task = {
  id: number;
  workflow_id: number;
  title: string;
  description: string | null;
  related_claim: string | null;
  owner: string | null;
  priority: TaskPriority;
  status: TaskStatus;
  created_at: string;
  updated_at: string | null;
};

export type TaskCreateInput = {
  workflow_id: number;
  title: string;
  description?: string;
  related_claim?: string;
  owner?: string;
  priority: TaskPriority;
  status: TaskStatus;
};

export type TaskUpdateInput = Partial<Omit<TaskCreateInput, 'workflow_id'>>;

export type DraftTask = {
  title: string;
  description: string;
  related_claim: string;
  owner: string;
  priority: TaskPriority;
  status: TaskStatus;
};

export type DraftTaskQueueItem = DraftTask & {
  source?: 'suggested_task' | 'idea_builder';
};

File: backend/app/schemas/task.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

from app.schemas.common import ORMModel


TaskPriority = Literal["low", "medium", "high"]
TaskStatus = Literal["todo", "doing", "done"]


class TaskCreate(BaseModel):
    workflow_id: int
    title: str = Field(..., max_length=255)
    description: str | None = None
    related_claim: str | None = None
    owner: str | None = Field(default=None, max_length=100)
    priority: TaskPriority = "medium"
    status: TaskStatus = "todo"


class TaskUpdate(BaseModel):
    title: str | None = Field(default=None, max_length=255)
    description: str | None = None
    related_claim: str | None = None
    owner: str | None = Field(default=None, max_length=100)
    priority: TaskPriority | None = None
    status: TaskStatus | None = None


class TaskRead(ORMModel):
    id: int
    workflow_id: int
    title: str
    description: str | None = None
    related_claim: str | None = None
    owner: str | None = None
    priority: str
    status: str
    created_at: datetime
    updated_at: datetime | None = None

